lib/spack/spack/util/module_cmd.py

```diff
@@ -41,6 +41,8 @@ def get_module_cmd():
 
 def get_module_cmd_from_which():
     module_cmd = which('modulecmd')
+    if not module_cmd:
+        raise ModuleError('`which` did not find any modulecmd executable.')
```

Comment:
Learn something new every day!

I don't love that error message for this spot, but I agree it's better. I'll change that.